pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -182,8 +182,8 @@ func (f *ring0Factory) DiscoveryClient() (discovery.CachedDiscoveryInterface, er
 	return f.discoveryFactory.DiscoveryClient()
 }
 
-func (f *ring0Factory) KubernetesClientSet() (*kubernetes.Clientset, error) {
-	return f.clientCache.KubernetesClientSetForVersion(nil)
+func (f *ring0Factory) KubernetesClientSet(requiredVersion *schema.GroupVersion) (*kubernetes.Clientset, error) {
+	return f.clientCache.KubernetesClientSetForVersion(requiredVersion)
```

Comment:
setting a GroupVersion for an external client does not do anything, compare https://github.com/sttts/kubernetes/blob/fd48a4a8959c272099aa3818ad94f25f39cfe43e/staging/src/k8s.io/client-go/kubernetes/typed/apps/v1/apps_client.go#L70. The GroupVersion in the rest client config is unconditionally overridden.